third_party/forked/gomod-collector/licenses.go

```diff
@@ -0,0 +1,203 @@
+// Copyright 2019 Google LLC
+// Copyright 2018 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/google/licenseclassifier"
+)
+
+var LicenseNames = []string{
+	"LICENCE",
+	"LICENSE",
+	"LICENSE.code",
+	"LICENSE.md",
+	"LICENSE.txt",
+	"COPYING",
+	"copyright",
+}
+
+const MatchThreshold = 0.9
+
+type LicenseFile struct {
+	Mod         Module
+	LicensePath string
+}
+
+func (lf *LicenseFile) Body() (string, error) {
+	body, err := ioutil.ReadFile(lf.LicensePath)
+	if err != nil {
+		return "", err
+	}
+	return string(body), nil
+}
+
+func (lt *LicenseFile) Classify(classifier *licenseclassifier.License) (string, error) {
+	body, err := lt.Body()
+	if err != nil {
+		return "", err
+	}
+	m := classifier.NearestMatch(body)
+	if m == nil {
+		return "", fmt.Errorf("unable to classify license: %v", lt.Mod)
+	}
+	return m.Name, nil
+}
+
+func (lt *LicenseFile) Check(classifier *licenseclassifier.License) error {
+	body, err := lt.Body()
+	if err != nil {
+		return err
+	}
+	ms := classifier.MultipleMatch(body, false)
+	for _, m := range ms {
+		return fmt.Errorf("Found matching forbidden license in %v: %v", lt.Mod, m.Name)
+	}
+	return nil
+}
+
+func (lt *LicenseFile) CSVRow(site string, classifier *licenseclassifier.License) (string, error) {
+	classification, err := lt.Classify(classifier)
+	if err != nil {
+		return "", err
+	}
+
+	return strings.Join([]string{
+		lt.Mod.Module,
+		"Static",
+		"", // TODO(mattmoor): Modifications?
```

Comment:
I don't think mattmoor would want his name here, should we open an issue for this?